src/core/src/pass/sdpa_to_paged_attention.cpp

```diff
@@ -36,12 +40,16 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
                     "the SDPAToPagedAttention transformation.");
 
     auto max_context_len = setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{}), "max_context_len");
-    ParameterVector model_remaining_params = {
+    ParameterVector model_remaining_params{
         setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}), "past_lens"),
         setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}), "subsequence_begins"),
-        setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}), "block_indices"),
         setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}), "block_indices_begins"),
     };
+    if (!m_use_block_indices_inputs) {
+        auto block_indices = setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}), "block_indices");
+        model_remaining_params.insert(model_remaining_params.begin() + 2, block_indices);
```

Comment:
why model_remaining_params.begin() + 2?

Reply:
Again, it has to be at a specific place in the parameters list